pkg/assembler/backends/testing/backend.go

```diff
@@ -62,19 +60,17 @@ type demoClient struct {
 	isDependency        []*model.IsDependency
 	certifyPkg          []*model.CertifyPkg
 	certifyVuln         []*model.CertifyVuln
-	hasSourceAt         []*model.HasSourceAt
 	certifyScorecard    []*model.CertifyScorecard
 	certifyBad          []*model.CertifyBad
 	isVulnerability     []*model.IsVulnerability
 	certifyVEXStatement []*model.CertifyVEXStatement
 	hasSLSA             []*model.HasSlsa
 	id                  uint32
+	index               indexType
```

Comment:
We have a global `index` and one in the client. We likely need to use only one of them, either the global (if we assume the client is singleton) or the one in the client struct (better for testing as we can then mock its contents).

Reply:
makes more sense to do it at the client struct. I will remove the duplicate.